src/components/Toolbar/Toolbar.tsx

```diff
@@ -62,32 +65,70 @@ export const Toolbar = ({ activeTab }: ToolbarProps) => {
   const cache = useAppSelector((app) => app.chat.cache);
   const { openSettings, openHotKeys } = useEventsBusForIDE();
 
-  const handleNavigation = (to: DropdownNavigationOptions | "chat") => {
-    if (to === "settings") {
-      openSettings();
-    } else if (to === "hot keys") {
-      openHotKeys();
-    } else if (to === "fim") {
-      dispatch(push({ name: "fill in the middle debug page" }));
-    } else if (to === "stats") {
-      dispatch(push({ name: "statistics page" }));
-    } else if (to === "restart tour") {
-      dispatch(restart());
-      dispatch(popBackTo({ name: "initial setup" }));
-      dispatch(push({ name: "welcome" }));
-    } else if (to === "integrations") {
-      dispatch(push({ name: "integrations page" }));
-    } else if (to === "chat") {
-      dispatch(popBackTo({ name: "history" }));
-      dispatch(push({ name: "chat" }));
-    }
-  };
+  const handleNavigation = useCallback(
+    (to: DropdownNavigationOptions | "chat") => {
+      if (to === "settings") {
+        openSettings();
+        void sendTelemetryEvent({
```

Comment:
would it be less code if if the middle ware listened for changes on the `pagesSlice` for navigation events, or do we want exclusively when they click the navigation ?

A similar approach could also be used for the `postMessage` function in `useEventBusForIDE`

I'm not sure if that's whats wanted though, could be a good idea for the future :)